File: java/src/main/java/com/allstask/models/MetricFilterOperator.java
// This file is @generated
package com.allstask.models;

import com.fasterxml.jackson.annotation.JsonValue;
import com.allstask.Utils.ToQueryParam;

public enum MetricFilterOperator implements ToQueryParam {
    EQUAL("EQUAL"),
    NOT_EQUAL("NOT_EQUAL"),
    IN("IN"),
    NOT_IN("NOT_IN");
    private final String value;

    MetricFilterOperator(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: typescript/src/models/metricFilterOperator.ts
// this file is @generated
/* eslint @typescript-eslint/no-explicit-any: 0 */
/**
* Operator of a pre-aggregation [`MetricFilter`]. `EQUAL`/`NOT_EQUAL` are the single-value
* forms of `IN`/`NOT_IN`. Negation (`NOT_EQUAL`/`NOT_IN`) is presence-required: an event
* missing the property is excluded.
*/
export enum MetricFilterOperator {
    Equal = 'EQUAL',
    NotEqual = 'NOT_EQUAL',
    In = 'IN',
    NotIn = 'NOT_IN',
    }

export const MetricFilterOperatorSerializer = {
    _fromJsonObject(object: any): MetricFilterOperator {
        return object;
    },

    _toJsonObject(self: MetricFilterOperator): any {
        return self;
    }
}